### DISCUSSION TITLE
fully static, zero JS pages (no React runtime, no rehydration) + Ability to bundle raw scripts (like Astro.js)

### DISCUSSION THREAD

**Discussion by UROjQ6r80p** [vercel/next.js#45174] [Ideas]
### Describe the feature you'd like to request

Fully static pages + ability to add custom <script> out of the box (without React runtime, hydration etc.)

I've seen this thread https://github.com/vercel/next.js/issues/5054 but from the responses there one could assume this is not really a big thing.

`Astro.js` proved that demand for this approach is quite big, and I think it's better to look at `Astro.js` popularity, and not a `gatsby-plugin-no-javascript` numbers.

### Describe the solution you'd like

I believe this sort of syntax out of the box would be a very useful feature, and that there is a growing demand for this approach.

`pages/index.js`
```
import NextRawScript from 'next/script';

function Index() {
    return (
        <div>
            <NextRawScript src="../anyfolder/script.js" />
            <h1>Hello, world!</h1>
            <button id="btn">Click me!</button>
        </div>
    )
}
```

`anyfolder/script.js`
```
// no React runtime, no rehydration
import axios from 'axios';

document.querySelector('#btn').addEventListener('click', e => axios.get(...), false);
```

possible output without React runtime, and fully tailored script bundle without React.
```
<head>
    <script defer src="script.5777c6664bc156.js"></script>
    ...
</head>
...
<body>
    <h1>Hello, world!</h1>
    <button id="btn">Click me!</button>
    // no rehydration = no need for __NEXT_DATA__ script
</body>
```

### Describe alternatives you've considered

Astro.js - not mature enough + ecosystem not mature enough.
unstable_runtimeJS option in Next.js - "unstable", no desire to add it into Next.js as a stable feature, and no ability to add custom scripts out of the box.

**Comment by CHAOSX0** [comment]
up this

**Comment by karpolan** [comment]
Something similar was achievable with `unstable_runtimeJS: false` but this config isn't supported for **App Router**.

So I double the request: we need a way to build **small static pages without any JS bundle** at all :)

**Comment by nicholasray** [comment]
I would love to have this!

**Comment by budishevskii** [comment]
I would also like this
